R/utils.R

```diff
@@ -230,3 +230,68 @@ padding_lst <- function(ft, indents) {
     flextable::padding(ft, s, 1, padding.left = (indents[s] + 1) * 10)
   }, seq_len(length(indents)), ft)
 }
+
+#' Split a text block into smaller blocks with a specified number of lines.
+#'
+#' This function takes a block of text and divides it into smaller blocks, each containing
+#' a specified number of lines.
+#'
+#' @param block_text A character vector containing the input block of text.
+#' @param n The number of lines per block.
+#'
+#' @return A list of character vectors, where each element is a smaller block of text
+#'         containing 'n' lines. If the input block of text has fewer lines than 'n', the
+#'         entire block is returned as a single element list.
+#'
+#' @keywords internal
+split_text_into_blocks <- function(block_text, n) {
+  lines <- strsplit(block_text, "\n")[[1]]
+  num_lines <- length(lines)
+
+  if (num_lines <= n) {
+    return(list(block_text))
+  }
+
+  num_blocks <- ceiling(num_lines / n)
+  blocks <- vector("list", length = num_blocks)
+
+  for (i in 1:num_blocks) {
+    start <- (i - 1) * n + 1
+    end <- min(i * n, num_lines)
+    block <- paste(lines[start:end], collapse = "\n")
+    blocks[[i]] <- block
+  }
+
+  return(blocks)
+}
```

Comment:
```suggestion
split_text_block <- function(x, n) {
  checkmate::assert_string(x)
  checkmate::assert_integerish(n, lower = 1L, len = 1L)
  
  lines <- strsplit(x, "\n")[[1]]
  
  if (length(lines) <= n) {
    return(x)
  }
  
  nblocks <- ceiling(length(lines) / n)
  ind <- rep(1:nblocks, each = n)[1:length(lines)]
  unname(tapply(lines, ind, paste, collapse = "\n"))
}
```

We try to avoid explicit return statements.